File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/BodaccNoticeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.Affaire;
import com.jss.osiris.modules.osiris.quotation.model.BodaccNotice;

public interface BodaccNoticeService {

    public void updateBodacNotices();

    public List<BodaccNotice> getBodaccNoticeByAffaire(Affaire affaire);

}

File: client/src/app/modules/quotation/services/bodacc-notice.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BodaccNotice } from '../model/BodaccNotice';

@Injectable({
  providedIn: 'root'
})
export class BodaccNoticeService extends AppRestService<BodaccNotice> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getBodaccNoticeForAffaire(affaireId: number) {
    return this.getList(new HttpParams().set("affaireId", affaireId), "bodacc-notice/affaire");
  }
}
